USWDS-Site: Add table accessibility tests page

amyleadem opened the pull request:
# Summary
- Added table accessibility tests page.
- Moved compliance-tag.html reference to styleguide.html
     - This allowed the compliance tag to be added to pages with the styleguide layout. 
     - Note: This is the same change added to #2540 

> [!Important]
> We need to confirm the changelog dates before merge.

## Resources
- [Creating accessibility checklists process doc](https://docs.google.com/document/d/14WTCsdzgwpoz_Miztd66RmEGfwwSEUFCVN1DkSQITTE/edit) (Google docs 🔒)
- [Accessibility checklist spreadsheet](https://docs.google.com/spreadsheets/d/1fAJaObQHv7aIaeUpH4fTLmauhmugMtNCrbpZe4BJcPY/edit#gid=97481116) (Google sheets 🔒)

## Related issue
Closes #2534 

## Preview link
- [Table - accessibility test page](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-table-checklist/components/table/accessibility-tests/)
- [Table main page](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-table-checklist/components/table/)

## Testing and review
1. Confirm that both the main component page and the accessibility tests page has the correct compliance tag at the top.
    1. Confirm that the compliance tag does NOT show up on pages that do not have a related a11y tests page.
1. Confirm that the main component page links to the accessibility tests page in the side navigation.
1. Confirm that the main component page links to the accessibility tests page in the accessibility section. 
    1. Confirm that this text references the correct component. For example "Test the icon in your own project." 
1. Check that accessibility tests page provides the correct counts for each test status type.
1. Confirm the test checklist summaries and data are accurate.  
1. Confirm no visual issues.
1. Confirm there is an appropriate changelog entry on both the main component page and the accessibility tests page.

amyleadem reviewed (commented):
@alex-hull @amycole501 @sarah-sch @finekatie 
This is ready for your review. I've added some comments below with some open questions. Please let me know if you have any questions or comments.

amyleadem commented on _data/accessibility-tests/table.yml:
```diff
@@ -0,0 +1,108 @@
+    status_details: |
+      The scrollable table does not scroll with keystrokes in JAWS.
+      A user would need to use their mouse to follow along with the focus indicator in JAWS.
+    test_type: keyboard
```
> [!Note]
> Note to self: need to add a link to the related issue.

amyleadem replied:
Added link to related issue in [a675795](https://github.com/uswds/uswds-site/pull/2538/commits/a6757952b668c94cf170bd2bf0213432dc134a50)

alex-hull commented:
@amyleadem It looks like the second "passes with exceptions" is not the yellow that the first one. It also does not look like it is in the table at the top of the page.

amyleadem commented on _data/accessibility-tests/table.yml:
```diff
@@ -0,0 +1,108 @@
+      When using a keyboard to navigate a table,
+      a user can navigate using arrow keys or the tab key with no unexpected cues or instruction.
+    test_status: exception
+    status_details: |
+      The scrollable table does not scroll with keystrokes in JAWS.
+      A user would need to use their mouse to follow along with the focus indicator in JAWS.
```
Update: Discussed this in chat and it is a confirmed issue. I have opened a [placeholder issue](https://github.com/uswds/uswds/issues/5828) for it and @amycole501 will add detail to the issue later today.

amyleadem commented:
@alex-hull The second "pass with exceptions" item is only showing up that way because it didn't have a test status in the spreadsheet. Can you respond to the questions in the [related comment above](https://github.com/uswds/uswds-site/pull/2538#discussion_r1531197527) about how to move forward with that bb?

amycole501 commented on _data/accessibility-tests/table.yml:
```diff
@@ -0,0 +1,108 @@
+    test_status: pass
+    test_type: screen_reader
+    version_tested: 3.6.0
+    wcag_criterion: 4.1.1
```
Agreed; remove it and if we find it's needed for future audits we can add it in with additional context.

amycole501 commented on _data/accessibility-tests/table.yml:
```diff
@@ -0,0 +1,108 @@
+      you can easily navigate in and out of the table cells using only keystrokes.
+    test_status:
+    test_type: keyboard
+    version_tested:
```
We sent an update that the table tests were retested or tested and just not updated in the spreadsheet. Let us know if there are any outliers.

amyleadem replied:
Thanks for the update. I've updated the test status here to `pass` in [e70b387](https://github.com/uswds/uswds-site/pull/2538/commits/e70b38746bf87c95c795461917065a87dae0cb31) to match the update in the spreadsheet. I also updated the version tested for all tests to 3.7.1 in [c5517e0](https://github.com/uswds/uswds-site/pull/2538/commits/c5517e0ff65d178e712ea1c6ef494d5cfaf771f5) to match the updated "last tested" date.

amycole501 commented on _data/accessibility-tests/table.yml:
Thank you for the comments I agree on all counts based on our discussion; we can circle back to adding those as we add further testing guidance and or nuance to the table component.

amycole501 commented on _data/accessibility-tests/table.yml:
```diff
@@ -0,0 +1,108 @@
+  - summary: Table cells are read in a logical order.
+    summary_additional: |
+      When using a screen reader to navigate a table,
+      you hear the table cells for an entire row read in order from left to right before moving down to the next row.
```
Perhaps "you hear the table cells for an entire row read in the order in which is logical for the language being used."

alex-hull replied:
Good point. I am wondering if reading style changes if a table is styled with items left to right. For example, header rows and columns could be on the left side, but does the table then flip to meet someone's needs that reads right to left.

alex-hull replied:
But I think how it has been worded appropriately above!

alex-hull replied:
Ok...I thought I replied to this, but it is now gone. I think the language is a good, broad way of putting it. However, if someone does read right to left with a screen reader, does the table then flip so it meets their needs? In my mind, the data cells would be read backwards and therefore make it a bit confusing.

amyleadem commented:
Update to the conversation above: I changed the status of the second "pass with exceptions" item to "passed" based on the recent update to the spreadsheet. Updated in [e70b387](https://github.com/uswds/uswds-site/pull/2538/commits/e70b38746bf87c95c795461917065a87dae0cb31)

amycole501 reviewed (approved):
I made comments but overall the page looks great! Nothing major of note to change.

amyleadem commented:
@sarah-sch Your edits should now be live. Let me know if you need any changes!

amyleadem commented on _data/accessibility-tests/table.yml:
```diff
@@ -0,0 +1,110 @@
+  - summary: Screen readers correctly announce table elements.
+    summary_additional: |
+      When using a screen reader to navigate a table,
+      the screen reader announces the table and calls the header rows "column headers".
```
> [!note]
> I did not incorporate this because we removed this test item in [e664553](https://github.com/uswds/uswds-site/pull/2538/commits/e66455340b2a0c2efae615d113213caf0b7645ad)

amyleadem commented on _data/accessibility-tests/table.yml:
```diff
@@ -0,0 +1,110 @@
+  - summary: The table has a header and it is associated with the correct table cells.
+    summary_additional: |
+      When using a screen reader to navigate a table,
+      you can clearly hear which column head the data falls under.
```
Updated in [911381c](https://github.com/uswds/uswds-site/pull/2538/commits/911381c728483d213466fbfc8db4cca5c27c501d)

amyleadem commented on _data/accessibility-tests/table.yml:
```diff
@@ -0,0 +1,110 @@
+    test_type: screen_reader
+    version_tested: 3.6.0
+    wcag_criterion: 2.1.1
+  - summary: The table has a header and it is associated with the correct table cells.
```
Updated in [911381c](https://github.com/uswds/uswds-site/pull/2538/commits/911381c728483d213466fbfc8db4cca5c27c501d)

amyleadem commented on _data/accessibility-tests/table.yml:
```diff
@@ -0,0 +1,110 @@
+# Zoom/screen magnification tests
+  - summary: Table content is visible and functional when screen is magnified.
+    summary_additional: |
+      When you zoom the screen to 200%,
```
Updated in [911381c](https://github.com/uswds/uswds-site/pull/2538/commits/911381c728483d213466fbfc8db4cca5c27c501d)

amyleadem commented on _data/accessibility-tests/table.yml:
```diff
@@ -0,0 +1,110 @@
+  - summary: Focus indicator is clear on interactive items in tables.
+    summary_additional: |
+      When using a keyboard to navigate a table,
+      a focus indicator appears around all interactive elements (e.g. arrows for sorting, links).
```
Updated in [911381c](https://github.com/uswds/uswds-site/pull/2538/commits/911381c728483d213466fbfc8db4cca5c27c501d)

amyleadem commented:
@sarah-sch Your requested updates are now live. Let me know if you need anything else.

sarah-sch reviewed (approved):
Approve as an MVP after fed review

amyleadem commented on _data/accessibility-tests/table.yml:
```diff
@@ -0,0 +1,102 @@
+      a user can navigate using arrow keys or the tab key with no unexpected cues or instruction.
+    test_status: exception
+    status_details: |
+      The scrollable table does not scroll with keystrokes in JAWS.
+      A user would need to use their mouse to follow along with the focus indicator in JAWS.
```
```suggestion
      The scrollable table does not scroll with left and right arrow keystrokes in JAWS and NVDA. 
      The offscreen table data is read aloud, but a user would need to use their mouse to follow along with the focus indicator.
```

amycole501 commented on _data/accessibility-tests/table.yml:
```diff
@@ -0,0 +1,135 @@
+    test_status: pass
+    test_type: screen_reader
+    version_tested: 3.6.0
+    wcag_criterion: 2.1.1
```
Agreed, it's OK to make that change. Headings and labels are typically geared more for the AT tests.

amyleadem commented:
@thisisdano @annepetersen This is ready for your review. FYI - I am able to get pa11y tests to pass locally, but they are still crashing here. Going to keep trying to get them to pass here 🏄
